File: tutorial/Ex03.java
// Ex03.java
// No fish class, overrides GameGrid.act() and GameGrid.reset()

import ch.aplu.jgamegrid.*;
import java.awt.Color;

public class Ex03 extends GameGrid
{
  private Actor nemo = new Actor("sprites/nemo.gif");

  public Ex03()
  {
    super(10, 10, 50, Color.red, "sprites/reef.gif");
    addActor(nemo, new Location(2, 4));
    show();
  }

  // Called in every simulation cycle
  public void act()
  {
    nemo.move();
    if (!nemo.isMoveValid())
    {
      nemo.turn(180);
      nemo.setHorzMirror(!nemo.isHorzMirror()) ;
    }
  }

  public static void main(String[] args)
  {
    new Ex03();
  }
}

File: python/GameGridEx/Ex03.py
# Ex03.py

from ch.aplu.jgamegrid import GameGrid, Actor, Location
from ch.aplu.util import X11Color

# ---------------- class MyGame -------------------
class MyGame(GameGrid):
   def __init__(self):
      GameGrid.__init__(self, 10, 10, 60, X11Color("red"), "sprites/reef.gif")
      self.addActor(nemo, Location(2, 4))
      self.show()

   def act(self):
      nemo.move()
      if not nemo.isMoveValid():
         nemo.turn(180)
         nemo.setHorzMirror(not nemo.isHorzMirror())
      
# ---------------- main --------------------------
nemo = Actor("sprites/nemo.gif")
MyGame()




